HW2/User.cpp

```diff
@@ -0,0 +1,83 @@
+//
+// Created by genom10 on 07.03.2021.
+//
+
+#include <iostream>
+#include <utility>
+#include "User.h"
+
+User::User(UserType::Type type, std::string name){
+    this->type = type;
+    this->name = std::move(name);
+}
+
+User::User(UserType::Type type, std::string name, std::string creator) : User::User(type, name) {
+    this->creator = std::move(creator);
+}
+
+UserType::Type User::getUserType() const {
+    return type;
+}
+
+void User::setUserType(UserType::Type userType) {
+    User::type = userType;
+}
+
+std::string User::getTextUserType() const{
+    switch(User::type){
+        case UserType::student : return "student";
+        case UserType::admin : return "admin";
+        case UserType::director : return "director";
+        case UserType::lab_employee : return "lab_employee";
+        case UserType::professor : return "professor";
+    }
+    return "unknown level";
+}
+
+std::string User::getTextUserType(UserType::Type userType) {
+    switch(userType){

```

Comment:
are you sure you need two copies of the implementation of the same method?

Reply:
One returns TextUserType of the instance and another returns TextUserType of given UserType